### DISCUSSION TITLE
Suddenly Stopped Working - 403 Forbidden

### DISCUSSION THREAD

**Discussion by jwj777** [tweepy/tweepy#2131] [Questions] [answered]
This script worked a week ago but I tried to run it today and got an error.

python - 3.10.7
tweepy - 4.7.0

Auth and request code:
```
# Initialize client
client = tweepy.Client(resources.bearer_token)

# My request
response = client.get_users(usernames=handle_list, user_fields=["id","public_metrics"])
```

The error:

WARNING:root:Exception Name: Forbidden
Traceback (most recent call last):
  File "c:\Users\webst\Dev\rank\venv\twitter_ds\twitter_reqs.py", line 39, in twitter_user_obj
    response = client.get_users(usernames=handle_list)
  File "c:\Users\webst\Dev\rank\venv\lib\site-packages\tweepy\client.py", line 1969, in get_users
    return self._make_request(
  File "c:\Users\webst\Dev\rank\venv\lib\site-packages\tweepy\client.py", line 118, in _make_request
    response = self.request(method, route, params=request_params,
  File "c:\Users\webst\Dev\rank\venv\lib\site-packages\tweepy\client.py", line 92, in request
    raise Forbidden(response)
tweepy.errors.Forbidden: 403 Forbidden

**Comment by jwj777** [comment]
Updated to 4.14

Slightly different error now:

WARNING:root:Exception Name: Forbidden
Traceback (most recent call last):
  File "c:\Users\webst\Dev\rank\venv\twitter_ds\twitter_reqs.py", line 39, in twitter_user_obj
    response = client.get_users(usernames=handle_list, user_fields=["id","public_metrics"])
  File "c:\Users\webst\Dev\rank\venv\lib\site-packages\tweepy\client.py", line 2501, in get_users
    return self._make_request(
  File "c:\Users\webst\Dev\rank\venv\lib\site-packages\tweepy\client.py", line 129, in _make_request
    response = self.request(method, route, params=request_params,
  File "c:\Users\webst\Dev\rank\venv\lib\site-packages\tweepy\client.py", line 100, in request
    raise Forbidden(response)
tweepy.errors.Forbidden: 403 Forbidden
When authenticating requests to the Twitter API v2 endpoints, you must use keys and tokens from a Twitter developer App that is attached to a Project. You can create a project via the developer portal.

I regenerated my bearer token from within the Twitter Developer Portal

**Comment by Harmon758** [comment]
Is your App attached to a Project?
What type/level of API access do you have?

**Comment by Aluerie** [comment]
Same here, free level. App is attached to a project. But `tweepy.asynchronous.AsyncStreamingClient` stopped working today with same error as OP. Looks like free users are now actually getting cut off.

**Comment by Harmon758** [comment]
Streaming was never available with Free access.
https://twittercommunity.com/t/announcing-new-access-tiers-for-the-twitter-api/188728
https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api

If you're referring to an old access plan, it looks like projects with those were changed to Free access, with any extra apps changed to standalone apps.
https://twittercommunity.com/t/v2-suddenly-getting-client-not-enrolled-today/195456
You can check your Developer Portal.

**Comment by SynthSy** [comment]
> Streaming was never available with Free access. https://twittercommunity.com/t/announcing-new-access-tiers-for-the-twitter-api/188728 https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api
> 
> If you're referring to an old access plan, it looks like projects with those were changed to Free access, with any extra apps changed to standalone apps. https://twittercommunity.com/t/v2-suddenly-getting-client-not-enrolled-today/195456 You can check your Developer Portal.

Previously, there were accounts that were grandfathered into an elevated free plan that included User Lookup, but I see now they've changed those accounts into their Free plan.

**Comment by Harmon758** [comment]
Yes, Elevated access would be one of the old access plans.
It was always free, so I would hardly say it was grandfathered in any way, but rather left active in a limbo state until now.

Regardless, without knowing the API access that they have, none of this is necessarily relevant to the original issue.

**Comment by Aluerie** [comment]
Looks like you're right for my case, old screenshots have "Elevated" access written in them while today it says "Free" access... How sad. Thanks for the info.

**Comment by jwj777** [comment]
I'm not sure what access I had before, but now it says "Free"

Did I get suddenly downgraded?  Does this mean I have to pay $100 per month now to do User follower lookups?

**Comment by jwj777** [comment]
Yeah, that was the deal. I was on some kind of old plan. F*ck.

**Comment by Harmon758** [comment, marked as answer]
> Did I get suddenly downgraded?

Yes, https://github.com/tweepy/tweepy/issues/2130#issuecomment-1585286109.

> Does this mean I have to pay $100 per month now to do User follower lookups?

Yes, you'll have to pay for access to do so now.

**Comment by cgivre** [comment]
This happened to me as well.  If I could offer a suggestion, could we include this somewhere in the docs so that people know that this feature doesn't work on free tiers?  (Until Elon decides to change it next week)
